### PR TITLE
Optimize code size (and speed) by using bit access instructions when accessing IO* registers

### PR DESCRIPTION
**Pull request by purdeaandrei**
Saves 33 bytes.

### CODE REVIEW

File: firmware/glasgow.h [hunk 1]
```diff
@@ -49,6 +49,28 @@ enum {
 #define PIND_ENVB             6
 #define PIND_OEQ_N_REVAB      7
 
+#define CONCAT(a, b) a ## b
```
Comment (whitequark) [inline firmware/glasgow.h:52]:
```suggestion
// Set up aliases for all of the GPIO pins accessible by `setb` and `clr`
// to reduce code size.
#define CONCAT(a, b) a ## b
```

File: firmware/main.c [hunk 2]
```diff
@@ -1007,14 +1007,14 @@ int main() {
...
         // blink. That's okay though.)
         switch (USBFRAMEH >> 1) {
-          case 0b00: IOD |=  (1<<PIND_LED_FX2); break;
-          case 0b01: IOD ^=  (1<<PIND_LED_FX2); break;
-          case 0b10: IOD &= ~(1<<PIND_LED_FX2); break;
-          case 0b11: IOD ^=  (1<<PIND_LED_FX2); break;
+          case 0b00: IO_LED_FX2 = 1; break;
+          case 0b10: IO_LED_FX2 = 0; break;
+          case 0b01:
+          case 0b11: IO_LED_FX2 ^= 1; break;
```
Comment (whitequark) [inline firmware/main.c:1013]: Does this result in a `CPL` being generated?

**Review by whitequark — CHANGES_REQUESTED** [review]
Thanks, the PR looks good.

It should add a comment indicating intent, I've left one in the suggestions.

The commit message should follow our guidelines.

Unless I'm missing something, I think mentioning speed in the commit message is misleading, as I don't think this commit has had any meaningful or even measurable impact on performance whatsoever.

File: firmware/main.c (hunk 2 above)
Comment (purdeaandrei) [reply firmware/main.c:1013]: yes, I see `cpl    _PD2` in the assembly output.

File: firmware/glasgow.h (hunk 1 above)
Comment (purdeaandrei) [reply firmware/glasgow.h:54]: commend added

**Comment by purdeaandrei** [comment]
I've reworked the commit message, but 50 characters is very hard without starting the abbreviate words, and I see numerous past commit messages go above it. Right now it's 53 chars, including the "firmware: " prefix, is that okay?

**Comment by whitequark** [comment]
That's fine, I think it actually doesn't start wrapping before 70 characters and I misremembered the exact number. Can you add ` (+33 bytes XRAM)` to the end so that it's visible in the shortlog? I've found that useful to see at a glance in the past.

**Comment by purdeaandrei** [comment]
> That's fine, I think it actually doesn't start wrapping before 72 characters and I misremembered the exact number.
> 
> Can you add ` (+33 bytes XRAM)` to the end so that it's visible in the shortlog? I've found that useful to see at a glance in the past.

Done, whole line length is now 70

**Comment by whitequark** [comment]
> Done, whole line length is now 70

Thanks! And yep, I misremembered, 50 is only a guideline, 72 is the actual wrapping limit. (Which I think is still too small and 80 or 100 would be a lot more reasonable, but oh well.)

**Review by whitequark — APPROVED** [review]

**Comment by whitequark** [comment]
Oh oops, I accidentally hit the "Update branch" button...

**Comment by purdeaandrei** [comment]
Should I rebase and force push  again?

**Comment by whitequark** [comment]
Yep, please!

**Comment by purdeaandrei** [comment]
done

**Comment by whitequark** [comment]
Thanks!

**Pull request merged**